src/bootstrapper.cc

```diff
@@ -116,6 +116,15 @@ void SetupPromises(const FunctionCallbackInfo<Value>& args) {
   env->set_promise_reject_handled_function(args[1].As<Function>());
 }
 
+void EnqueueMicrotask(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsFunction());
+
+  isolate->EnqueueMicrotask(args[0].As<Function>());
+}
```

Comment:
How does this relates to async_hooks? I think we would lose context if we do nothing.
cc @nodejs/diagnostics

Reply:
echoing this... it definitely needs to be possible to track a queued microtask with async_hooks... *especially* if the intent of this is to eventually replace `process.nextTick()` for most use cases.